% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\usepackage{amsmath,amssymb}
\usepackage{lmodern}
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\hypersetup{
  pdftitle={Scientific Programming Project},
  pdfauthor={Thales A. de M. Fernandes},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}
\urlstyle{same} % disable monospaced font for URLs
\usepackage[margin=1in]{geometry}
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering
\newlength{\cslhangindent}
\setlength{\cslhangindent}{1.5em}
\newlength{\csllabelwidth}
\setlength{\csllabelwidth}{3em}
\newlength{\cslentryspacingunit} % times entry-spacing
\setlength{\cslentryspacingunit}{\parskip}
\newenvironment{CSLReferences}[2] % #1 hanging-ident, #2 entry spacing
 {% don't indent paragraphs
  \setlength{\parindent}{0pt}
  % turn on hanging indent if param 1 is 1
  \ifodd #1
  \let\oldpar\par
  \def\par{\hangindent=\cslhangindent\oldpar}
  \fi
  % set entry spacing
  \setlength{\parskip}{#2\cslentryspacingunit}
 }%
 {}
\usepackage{calc}
\newcommand{\CSLBlock}[1]{#1\hfill\break}
\newcommand{\CSLLeftMargin}[1]{\parbox[t]{\csllabelwidth}{#1}}
\newcommand{\CSLRightInline}[1]{\parbox[t]{\linewidth - \csllabelwidth}{#1}\break}
\newcommand{\CSLIndent}[1]{\hspace{\cslhangindent}#1}
\ifLuaTeX
  \usepackage{selnolig}  % disable illegal ligatures
\fi

\title{Scientific Programming Project}
\author{Thales A. de M. Fernandes}
\date{2022-08-15}

\begin{document}
\maketitle

\hypertarget{introduction}{%
\subsection{1. Introduction}\label{introduction}}

The detailed comparison of homologous protein structures can be used to
infer pathways for evolutionary process, mechanisms for conformational
changes, dynamics, internal motions and intrinsic dynamics, structure
rearrangements, and conformational and allosteric mechanisms. The
increasing availability of high-resolution protein structures allows the
integration of molecular structure, dynamics and evolution analysis.

In this context, this project aimed to process, organize and explore
structure and sequence data using \texttt{bio3d} package Grant et al.
(2006) Skjærven et al. (2014) Grant, Skjærven, and Yao (2021). The goals
of this project are read and write structures, perform atom summaries,
atom selection, re-orientation, superposition, clustering, distance
matrix analysis, structure and sequence conservation analysis, and
principal component analysis (PCA) from Protein Data Bank (PDB) database
Berman et al. (2000).

\hypertarget{exploratory-data-analysis-of-protein-structure-and-sequence}{%
\subsection{2. Exploratory data analysis of protein structure and
sequence}\label{exploratory-data-analysis-of-protein-structure-and-sequence}}

The (PBD) is a database of protein structures grouped in .PDB files,
which contains a list of all atoms, the x, y, z coordinates and
secondary structure elements.

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\#Read a PDB file }
\NormalTok{pdb }\OtherTok{\textless{}{-}} \FunctionTok{read.pdb}\NormalTok{(}\StringTok{"4q21"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   Note: Accessing on-line PDB file
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{print}\NormalTok{(pdb)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
##  Call:  read.pdb(file = "4q21")
## 
##    Total Models#: 1
##      Total Atoms#: 1447,  XYZs#: 4341  Chains#: 1  (values: A)
## 
##      Protein Atoms#: 1340  (residues/Calpha atoms#: 168)
##      Nucleic acid Atoms#: 0  (residues/phosphate atoms#: 0)
## 
##      Non-protein/nucleic Atoms#: 107  (residues: 80)
##      Non-protein/nucleic resid values: [ GDP (1), HOH (78), MG (1) ]
## 
##    Protein sequence:
##       MTEYKLVVVGAGGVGKSALTIQLIQNHFVDEYDPTIEDSYRKQVVIDGETCLLDILDTAG
##       QEEYSAMRDQYMRTGEGFLCVFAINNTKSFEDIHQYREQIKRVKDSDDVPMVLVGNKCDL
##       AARTVESRQAQDLARSYGIPYIETSAKTRQGVEDAFYTLVREIRQHKL
## 
## + attr: atom, xyz, seqres, helix, sheet,
##         calpha, remark, call
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{attributes}\NormalTok{(pdb)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## $names
## [1] "atom"   "xyz"    "seqres" "helix"  "sheet"  "calpha" "remark" "call"  
## 
## $class
## [1] "pdb" "sse"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{head}\NormalTok{(pdb}\SpecialCharTok{$}\NormalTok{atom)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   type eleno elety  alt resid chain resno insert      x      y      z o     b
## 1 ATOM     1     N <NA>   MET     A     1   <NA> 64.080 50.529 32.509 1 28.66
## 2 ATOM     2    CA <NA>   MET     A     1   <NA> 64.044 51.615 33.423 1 29.19
## 3 ATOM     3     C <NA>   MET     A     1   <NA> 63.722 52.849 32.671 1 30.27
## 4 ATOM     4     O <NA>   MET     A     1   <NA> 64.359 53.119 31.662 1 34.93
## 5 ATOM     5    CB <NA>   MET     A     1   <NA> 65.373 51.805 34.158 1 28.49
## 6 ATOM     6    CG <NA>   MET     A     1   <NA> 65.122 52.780 35.269 1 32.18
##   segid elesy charge
## 1  <NA>     N   <NA>
## 2  <NA>     C   <NA>
## 3  <NA>     C   <NA>
## 4  <NA>     O   <NA>
## 5  <NA>     C   <NA>
## 6  <NA>     C   <NA>
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{plot.bio3d}\NormalTok{(pdb}\SpecialCharTok{$}\NormalTok{atom}\SpecialCharTok{$}\NormalTok{b[pdb}\SpecialCharTok{$}\NormalTok{calpha], }\AttributeTok{sse=}\NormalTok{pdb, }\AttributeTok{typ=}\StringTok{"l"}\NormalTok{, }\AttributeTok{ylab=}\StringTok{"B{-}factor"}\NormalTok{) }\CommentTok{\#Plotting the B{-}factor}
\end{Highlighting}
\end{Shaded}

\includegraphics{PROJECT_PDF_files/figure-latex/unnamed-chunk-2-1.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\#Select and write a PDB object}

\NormalTok{b.inds }\OtherTok{\textless{}{-}} \FunctionTok{atom.select}\NormalTok{(pdb, }\StringTok{"back"}\NormalTok{) }\CommentTok{\#selecting the backbone atoms}
\NormalTok{backpdb }\OtherTok{\textless{}{-}} \FunctionTok{trim.pdb}\NormalTok{(pdb, b.inds) }\CommentTok{\#creating a new object of backbone atoms}
\FunctionTok{write.pdb}\NormalTok{(backpdb, }\AttributeTok{file=}\StringTok{"\textasciitilde{}/project/data/processed/4q21\_back.pdb"}\NormalTok{) }\CommentTok{\#write a .PDB file}

\CommentTok{\#Identification of dynamic domains }
\NormalTok{pdb }\OtherTok{\textless{}{-}} \FunctionTok{read.pdb}\NormalTok{(}\StringTok{"1d1d"}\NormalTok{, }\AttributeTok{multi=}\ConstantTok{TRUE}\NormalTok{) }\CommentTok{\# Read multi{-}model PDB file}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   Note: Accessing on-line PDB file
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{gs }\OtherTok{\textless{}{-}} \FunctionTok{geostas}\NormalTok{(pdb)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   .. 220 'calpha' atoms selected
##   .. 'xyz' coordinate data with 20 frames 
##   .. 'fit=TRUE': running function 'core.find'
##   .. coordinates are superimposed to core region
##   .. calculating atomic movement similarity matrix ('amsm.xyz()') 
##   .. dimensions of AMSM are 220x220
##   .. clustering AMSM using 'kmeans' 
##   .. converting indices to match input 'pdb' object 
##      (additional attribute 'atomgrps' generated)
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{plot}\NormalTok{(gs)}
\end{Highlighting}
\end{Shaded}

\includegraphics{PROJECT_PDF_files/figure-latex/unnamed-chunk-2-2.pdf}

\hypertarget{comparative-analysis-of-protein-structures}{%
\subsection{3. Comparative analysis of protein
structures}\label{comparative-analysis-of-protein-structures}}

\hypertarget{structure-search-and-selection}{%
\subsubsection{3.1. Structure search and
selection}\label{structure-search-and-selection}}

From the protein sequence or database identifier, related structures can
be collected from the PDB database.

In the example below, structures correlated with protein kinase A (PDB:
1L3R) are identified, annotated and downloaded.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{aa }\OtherTok{\textless{}{-}} \FunctionTok{get.seq}\NormalTok{(}\StringTok{"1L3R\_E"}\NormalTok{, }\AttributeTok{outfile =} \StringTok{"\textasciitilde{}/project/data/raw/seqs.fasta"}\NormalTok{) }\CommentTok{\#Obtain the protein sequence}

\NormalTok{blast }\OtherTok{\textless{}{-}} \FunctionTok{blast.pdb}\NormalTok{(aa) }\CommentTok{\#find similar sequences using BLAST sequence alignment algorithm}

\NormalTok{hits }\OtherTok{\textless{}{-}} \FunctionTok{plot}\NormalTok{(blast) }\CommentTok{\#Plot the BLAST results}
\end{Highlighting}
\end{Shaded}

\includegraphics{PROJECT_PDF_files/figure-latex/unnamed-chunk-3-1.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{annotation }\OtherTok{\textless{}{-}} \FunctionTok{pdb.annotate}\NormalTok{(hits) }\CommentTok{\#Annotate the BLAST results into the variable "annotation"}

\NormalTok{pdb.id }\OtherTok{\textless{}{-}} \FunctionTok{with}\NormalTok{(annotation, }\FunctionTok{subset}\NormalTok{(hits}\SpecialCharTok{$}\NormalTok{pdb.id,resolution}\SpecialCharTok{\textless{}=}\DecValTok{3}\NormalTok{)) }\CommentTok{\#Filtering structures by resolution \textless{} 3}

\NormalTok{files }\OtherTok{\textless{}{-}} \FunctionTok{get.pdb}\NormalTok{(pdb.id, }\AttributeTok{path=}\StringTok{"\textasciitilde{}/project/data/raw"}\NormalTok{, }\AttributeTok{split=}\ConstantTok{TRUE}\NormalTok{) }\CommentTok{\#Download protein structures and split into single chains}
\end{Highlighting}
\end{Shaded}

\hypertarget{multiple-sequence-alignment}{%
\subsubsection{3.2. Multiple Sequence
Alignment}\label{multiple-sequence-alignment}}

In this step, the selected structures are submitted to multiple sequence
alignment algorithm using \texttt{MSA} package Bodenhofer et al. (2015).

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pdbs }\OtherTok{\textless{}{-}} \FunctionTok{pdbaln}\NormalTok{(files, }\AttributeTok{exefile =} \StringTok{"msa"}\NormalTok{, }\AttributeTok{outfile =} \StringTok{"aln.fa"}\NormalTok{) }\CommentTok{\# Align protein structures present in files variable}
\FunctionTok{plot}\NormalTok{(pdbs)}
\end{Highlighting}
\end{Shaded}

\includegraphics{PROJECT_PDF_files/figure-latex/unnamed-chunk-4-1.pdf}

\hypertarget{structure-fitting-and-analysis}{%
\subsubsection{3.3. Structure Fitting and
analysis}\label{structure-fitting-and-analysis}}

In this section, the aligned structures are superposed based on the
invariant structure core. The analysis involves individual residue
fluctuations (RMSF) and overall structural deviations (RMSD).

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{cores }\OtherTok{\textless{}{-}} \FunctionTok{core.find}\NormalTok{(pdbs) }\CommentTok{\#Find the core in the aligned structures}
\NormalTok{xyz }\OtherTok{\textless{}{-}} \FunctionTok{pdbfit}\NormalTok{(pdbs, }\AttributeTok{inds=}\NormalTok{cores) }\CommentTok{\#Obtain the coordinates of the core}
\end{Highlighting}
\end{Shaded}

\hypertarget{principal-component-analysis}{%
\subsubsection{3.4. Principal Component
Analysis}\label{principal-component-analysis}}

In this step, the superimposed structures are compared using principal
component analysis (PCA). The PCA correspond to a powerful multivariate
statistical technique to reduce the dimensionality of complex data into
principal components (PCs). The PCA highlights the patterns and
relationships present in protein structures transforming atomic
coordinates into PCs that represent directions where the structures
displays largest variances.

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\#Apply PCA for the aligned and non{-}gaped structures}
\NormalTok{pc }\OtherTok{\textless{}{-}} \FunctionTok{pca}\NormalTok{(xyz, }\AttributeTok{rm.gaps=}\ConstantTok{TRUE}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## NOTE: Removing 121 gap positions with missing coordinate data
##       retaining 284 non-gap positions for analysis.
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\#Structure clustering in the PC1{-}PC2 plane}
\NormalTok{d }\OtherTok{\textless{}{-}} \FunctionTok{dist}\NormalTok{(pc}\SpecialCharTok{$}\NormalTok{z[, }\DecValTok{1}\SpecialCharTok{:}\DecValTok{2}\NormalTok{]) }\CommentTok{\#Specifying the PCs}
\NormalTok{hc }\OtherTok{\textless{}{-}} \FunctionTok{hclust}\NormalTok{(d) }\CommentTok{\#Clustering}
\FunctionTok{hclustplot}\NormalTok{(hc, }\AttributeTok{k=}\DecValTok{2}\NormalTok{, }\AttributeTok{labels=}\NormalTok{pdb.id) }\CommentTok{\#Plotting the clustering}
\end{Highlighting}
\end{Shaded}

\includegraphics{PROJECT_PDF_files/figure-latex/unnamed-chunk-6-1.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{grps }\OtherTok{\textless{}{-}} \FunctionTok{cutree}\NormalTok{(hc, }\AttributeTok{k=}\DecValTok{2}\NormalTok{)}

\CommentTok{\#Generate conformer plot, scree plot, and residue loadings plot}
\FunctionTok{plot}\NormalTok{(pc, }\AttributeTok{col=}\NormalTok{grps) }\CommentTok{\#Plotting the PCA}
\end{Highlighting}
\end{Shaded}

\includegraphics{PROJECT_PDF_files/figure-latex/unnamed-chunk-6-2.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{plot.bio3d}\NormalTok{(pc}\SpecialCharTok{$}\NormalTok{au[, }\DecValTok{1}\NormalTok{], }\AttributeTok{sse=}\FunctionTok{pdbs2sse}\NormalTok{(pdbs), }\AttributeTok{ylab=}\StringTok{"Loadings"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\includegraphics{PROJECT_PDF_files/figure-latex/unnamed-chunk-6-3.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\#Saving a  trajectory showing the collective motion defined by PC1}
\FunctionTok{mktrj}\NormalTok{(pc, }\AttributeTok{pc=}\DecValTok{1}\NormalTok{, }\AttributeTok{file=}\StringTok{"\textasciitilde{}/project/data/processed/pc\_1.pdb"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\hypertarget{document-and-session-details}{%
\subsection{4. Document and session
details}\label{document-and-session-details}}

These document represent the project from the Scientific Programming
curse from Serrapilheira/ICTP-SAIFR Training Program in Quantitative
Biology and Ecology

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{sessionInfo}\NormalTok{()}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## R version 4.2.0 (2022-04-22 ucrt)
## Platform: x86_64-w64-mingw32/x64 (64-bit)
## Running under: Windows 10 x64 (build 22000)
## 
## Matrix products: default
## 
## locale:
## [1] LC_COLLATE=Portuguese_Brazil.utf8  LC_CTYPE=Portuguese_Brazil.utf8   
## [3] LC_MONETARY=Portuguese_Brazil.utf8 LC_NUMERIC=C                      
## [5] LC_TIME=Portuguese_Brazil.utf8    
## 
## attached base packages:
## [1] stats4    stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
## [1] msa_1.28.0          Biostrings_2.64.0   GenomeInfoDb_1.32.2
## [4] XVector_0.36.0      IRanges_2.30.0      S4Vectors_0.34.0   
## [7] BiocGenerics_0.42.0 bio3d_2.4-3        
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_1.0.9             rstudioapi_0.13        knitr_1.39            
##  [4] magrittr_2.0.3         zlibbioc_1.42.0        lattice_0.20-45       
##  [7] R6_2.5.1               rlang_1.0.4            fastmap_1.1.0         
## [10] httr_1.4.3             highr_0.9              stringr_1.4.0         
## [13] tools_4.2.0            parallel_4.2.0         grid_4.2.0            
## [16] xfun_0.31              cli_3.3.0              htmltools_0.5.3       
## [19] yaml_2.3.5             digest_0.6.29          crayon_1.5.1          
## [22] GenomeInfoDbData_1.2.8 bitops_1.0-7           curl_4.3.2            
## [25] RCurl_1.98-1.7         evaluate_0.15          rmarkdown_2.14        
## [28] stringi_1.7.8          compiler_4.2.0         jsonlite_1.8.0
\end{verbatim}

\hypertarget{references}{%
\subsection*{5. References}\label{references}}
\addcontentsline{toc}{subsection}{5. References}

\hypertarget{refs}{}
\begin{CSLReferences}{1}{0}
\leavevmode\vadjust pre{\hypertarget{ref-berman2000protein}{}}%
Berman, Helen M, John Westbrook, Zukang Feng, Gary Gilliland, Talapady N
Bhat, Helge Weissig, Ilya N Shindyalov, and Philip E Bourne. 2000.
{``The Protein Data Bank.''} \emph{Nucleic Acids Research} 28 (1):
235--42.

\leavevmode\vadjust pre{\hypertarget{ref-bodenhofer2015msa}{}}%
Bodenhofer, Ulrich, Enrico Bonatesta, Christoph Horejš-Kainrath, and
Sepp Hochreiter. 2015. {``Msa: An r Package for Multiple Sequence
Alignment.''} \emph{Bioinformatics} 31 (24): 3997--99.

\leavevmode\vadjust pre{\hypertarget{ref-grant2006bio3d}{}}%
Grant, Barry J, Ana PC Rodrigues, Karim M ElSawy, J Andrew McCammon, and
Leo SD Caves. 2006. {``Bio3d: An r Package for the Comparative Analysis
of Protein Structures.''} \emph{Bioinformatics} 22 (21): 2695--96.

\leavevmode\vadjust pre{\hypertarget{ref-grant2021bio3d}{}}%
Grant, Barry J, Lars Skjærven, and Xin-Qiu Yao. 2021. {``The Bio3D
Packages for Structural Bioinformatics.''} \emph{Protein Science} 30
(1): 20--30.

\leavevmode\vadjust pre{\hypertarget{ref-skjaerven2014integrating}{}}%
Skjærven, Lars, Xin-Qiu Yao, Guido Scarabelli, and Barry J Grant. 2014.
{``Integrating Protein Structural Dynamics and Evolutionary Analysis
with Bio3D.''} \emph{BMC Bioinformatics} 15 (1): 1--11.

\end{CSLReferences}

\end{document}
